dtclient/ruxit_conf.go

```diff
@@ -0,0 +1,87 @@
+package dtclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	ruxit "github.com/Dynatrace/dynatrace-operator/conf"
+	"github.com/pkg/errors"
+)
+
+const notModifiedStatusCode = 304
+
+type RuxitProcResponse struct {
+	Revision   uint            `json:"revision"`
+	Properties []RuxitProperty `json:"properties"`
+}
+
+type RuxitProperty struct {
+	Section string `json:"section"`
+	Key     string `json:"key"`
+	Value   string `json:"value"`
+}
+
+func (rc RuxitProcResponse) ToMap() ruxit.ConfMap {
+	sections := map[string]map[string]string{}
+	for _, prop := range rc.Properties {
+		section := sections[prop.Section]
+		if section == nil {
+			section = map[string]string{}
+		}
+		section[prop.Key] = prop.Value
+		sections[prop.Section] = section
+	}
+	return sections
+}
+
+func (dtc *dynatraceClient) GetRuxitProcConf(prevRevision uint) (*RuxitProcResponse, error) {
+	ruxitConfURL := fmt.Sprintf("%s/v1/deployment/installer/agent/processmoduleconfig", dtc.url)
+
+	req, err := http.NewRequest("GET", ruxitConfURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing http request: %w", err)
+	}
+	query := req.URL.Query()
+	query.Add("revision", strconv.FormatUint(uint64(prevRevision), 10))
```

Comment:
what happens when there is no revision yet? can we drop this parameter when there is no `prevRevision`?

Reply:
I never saw that :D
it was always around 6000, and by defualt we call it with 0 so if they don't have anything we shouldn't get anything and we don't do anything :)

Reply:
i guess that's fine since 0 isn't a valid revision